Materials for "How to List the Files in a Directory with Python"

iansedano opened the pull request:
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

lpozo commented on python-get-all-files-in-directory/listdir.py:
```diff
@@ -0,0 +1,21 @@
+import contextlib
```
Do we need this file? I mean, it's not an example we used in the article.

lpozo commented on python-get-all-files-in-directory/scandir.py:
```diff
@@ -0,0 +1,18 @@
+import os
```
Do we need this file?

lpozo reviewed (changes requested):
Hey @iansedano!

I left a few comments below.

My main concern with this set of sample resources is they don't really contain the article's examples, which was... surprising.

Some of the examples introduce complex topics and tools that aren't covered in the article.

I also believe that we should simplify the performant testing scripts because they're not the main resource this folder should have.

I'd recommend simplifying these sample resources and making them reflect the article's content in the first place. Otherwise, we'll be introducing topics and examples without a good explanation and without any handholding. That may be unfair to our readers.

Please, let me know if you have any thoughts about my comments.

Take care!

iansedano commented on python-get-all-files-in-directory/glob.py:
```diff
@@ -0,0 +1,15 @@
+import pathlib
+from pprint import pp
+
+cwd = pathlib.Path("My Documents")
```
I've changed it to be `desktop` in the end

iansedano commented on python-get-all-files-in-directory/glob.py:
```diff
@@ -0,0 +1,15 @@
+pp(list(cwd.glob("**/*.py")))
+pp(list(cwd.rglob("*.py")))
+
+pp([file for file in cwd.glob("**/*") if file.suffix in [".py", ".md"]])
```
I've reworked this file to include most of the examples of the article.

iansedano commented:
Thanks for the review!

You are right that most of the materials are somewhat "extras" so I've moved most of them into a "bonus" folder.

I've simplified the performance script to include in the article, but have left the larger tests in the "bonus" folder.

Since the examples in the article are reasonably short, I thought it would be nice to include a bunch of bonus materials. I've been doing this a fair bit for these shorter articles. It just represents an added bonus to the readers. But I think you are right that not signposting this appropriately could be confusing, so I've also signposted this in the README.

lpozo reviewed (changes requested):
Hey @iansedano!

I left you a few comments below. I wonder if we need the `Desktop_large` directory. This commit will add 1.974 files to the repo.

You need to fix the linter broken check.

Thanks for updating this!

gahjelle reviewed (approved):
:+1: 

We may want to revisit `Desktop_large`. Instead of including all files in the repo, we could include a small script that recreates the directory.

iansedano commented:
Thanks @gahjelle, I've made another PR with those changes: https://github.com/realpython/materials/pull/335

There is already a module that generates folder structures, so I've added a quick note to that in the README.
